cmd/thanos/compact.go

```diff
@@ -746,6 +747,10 @@ func (cc *compactConfig) registerFlag(cmd extkingpin.FlagClause) {
 		"NOTE: This flag is ignored and (enabled) when --deduplication.replica-label flag is set.").
 		Hidden().Default("false").BoolVar(&cc.enableVerticalCompaction)
 
+	cmd.Flag("compact.group-concurrency", "The number of concurrent compactions from a single compaction group inside one compaction iteration. "+
```

Comment:
1. Can we add this flag next to `compact.concurrency` for readability? 
2. Are we sure we need this granuality of config? Is there a way to automatically detect this from `compact.concurrency`?